### PR TITLE
ft: add parent/generic modal component 📻

### LINKED ISSUE #53: Modal

### PR DESCRIPTION
**Pull request by shyakadavis**
# Description

This P.R deliver a parent modal component.

The `<Modal/>` component can take different props, including, but not limited to, a `header, body, footer` all which are html elements, and different actions and states to control visibility.

[Loom](https://www.loom.com/share/3780c9b0c02c4a99961d814ceb9ff710)

Fixes #53 

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have created a loom video for the new feature or enhancement
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by SergeInGithub — COMMENTED** [review]
Nice work here. 
Some ui suggestions though, i think the design could do with some improvement, mostly the close button, i think if it would just be the `X` icon without the border it'd look better

**Comment by SergeInGithub** [comment]
> Nice work here.
> Some ui suggestions though, i think the design could do with some improvement, mostly the close button, i think if it would just be the `X` icon without the border it'd look better

**Review by SergeInGithub — COMMENTED** [review]
Something else, I also noticed that `autodocs` on storybook aren't working properly for both the navbar or this component, maybe you should fix it or remove it since you already provide the `default` story

**Review by emilenfc — DISMISSED** [review]
Nice work

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Review by musllim — APPROVED** [review]
LGTM

**Review by mucyoalexaxel — DISMISSED** [review]
Great Work

**Review by ISHIMWE-Aime — DISMISSED** [review]
LGTM

File: src/components/shared/modals/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    if (disabled) return;
+    onSubmit();
+  }, [disabled, onSubmit]);
+
+  const handleSecondaryAction = useCallback(() => {
+    if (disabled || !secondaryAction) return;
+    secondaryAction();
+  }, [disabled, secondaryAction]);
+
+  if (!isOpen) return null;
```
Comment (mucyoalexaxel) [inline src/components/shared/modals/Modal.tsx:87]:
```suggestion
```

@shyakadavis Wrap all that logic into a custom hook,.

**Review by mucyoalexaxel — CHANGES_REQUESTED** [review]
@shyakadavis Review Requested changes

File: src/components/shared/modals/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                  colorScheme="btn-secondary"
+                />
+              </div>
+              {footer}
+            </footer>
+          </form>
+        </div>
+      </div>
+    </div>
+  );
```
Comment (mucyoalexaxel) [inline src/components/shared/modals/Modal.tsx:100]:
```suggestion
```
@shyakadavis Refactor this as well

**Review by mucyoalexaxel — CHANGES_REQUESTED** [review]
@shyakadavis Great Work here, Check some suggested change below

**Review by mucyoalexaxel — APPROVED** [review]
Great Job @shyakadavis

**Pull request merged**